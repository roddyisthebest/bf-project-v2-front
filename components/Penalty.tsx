import React from 'react';
import styled from 'styled-components/native';
import {PenaltyType} from '../types/Penalty';

const Container = styled.View`
  padding: 25px 30px;
  background-color: white;
`;

const Column = styled.View`
  flex-direction: row;
  justify-content: space-between;
`;

const Key = styled.Text`
  font-size: 15px;
  font-weight: 800;
  color: black;
`;

const Value = styled.Text`
  font-size: 15px;
  font-weight: 800;
  color: #687684;
`;

const Penalty = ({data}: {data: PenaltyType}) => {
  return (
    <Container>
      <Column style={{marginBottom: 12.5}}>
        <Key>금액</Key>
        <Value>{data.paper}</Value>
      </Column>
      <Column>
        <Key>날짜</Key>
        <Value>{data.weekend}</Value>
      </Column>
    </Container>
  );
};

export default Penalty;
