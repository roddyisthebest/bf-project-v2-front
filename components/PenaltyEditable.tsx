import React, {useCallback, useEffect, useState} from 'react';
import Icon from 'react-native-vector-icons/Ionicons';
import styled from 'styled-components/native';
import {LoggedInParamList} from '../navigation/Root';
import {NavigationProp, useNavigation} from '@react-navigation/native';
import {User} from '../types/User';
import {ActivityIndicator, Alert} from 'react-native';
import {checkPayed} from '../api/user';

const Container = styled.View`
  padding: 25px 30px;
  background-color: white;
`;

const UserBtn = styled.TouchableOpacity`
  flex-direction: row;
  margin-bottom: 20px;
  align-items: center;
`;
const UserImg = styled.ImageBackground`
  width: 30px;
  height: 30px;
  margin-right: 15px;
`;

const UserName = styled.Text`
  font-size: 22.5px;
  font-weight: 800;
  color: black;
`;

const Column = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const Key = styled.Text`
  font-size: 15px;
  font-weight: 800;
  color: black;
`;

const Value = styled.Text`
  font-size: 15px;
  font-weight: 800;
  color: #687684;
`;

const Btn = styled.TouchableOpacity<{bkgColor: string}>`
  width: 30px;
  height: 30px;
  border-radius: 20px;
  background-color: ${props => props.bkgColor};
  align-items: center;
  justify-content: center;
`;

const PenaltyEditable = ({data}: {data: User}) => {
  const navigation = useNavigation<NavigationProp<LoggedInParamList>>();

  const [check, setCheck] = useState<boolean>(true);
  const [loading, setLoading] = useState<boolean>(false);
  const setPayed = useCallback(async (id: number, payed: boolean) => {
    try {
      setLoading(true);
      await checkPayed(id, !payed);
      setCheck(prev => !prev);
    } catch (e) {
      console.log(e);
      Alert.alert('오류입니다.');
    } finally {
      setLoading(false);
    }
  }, []);
  useEffect(() => {
    setCheck(data.payed);
  }, [data]);
  return (
    <Container>
      <UserBtn
        onPress={() => {
          navigation.navigate('Stack', {
            screen: 'Detail',
            params: {id: data.id, uri: null},
          });
        }}>
        <UserImg
          source={{
            uri: data.img,
          }}
          resizeMode="cover"
          borderRadius={50}
        />
        <UserName>{data.name}</UserName>
      </UserBtn>
      <Column style={{marginBottom: 12.5, marginTop: 5}}>
        <Key>금액</Key>
        <Value>{data.Penalties[0].paper}</Value>
      </Column>
      <Column>
        <Key>payed</Key>
        <Btn
          bkgColor={check ? '#DFF3F1' : '#ffeaed'}
          onPress={() => {
            setPayed(data.id, check);
          }}
          disabled={loading}>
          {loading ? (
            <ActivityIndicator color={check ? '#10DDC2' : 'red'} size={15} />
          ) : (
            <Icon
              name={check ? 'checkmark-outline' : 'close-outline'}
              color={check ? '#10DDC2' : 'red'}
              size={18}
            />
          )}
        </Btn>
      </Column>
    </Container>
  );
};

export default PenaltyEditable;
