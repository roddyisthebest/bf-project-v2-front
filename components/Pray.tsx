import React from 'react';
import styled from 'styled-components/native';
import {PrayType} from '../types/Pray';

const Container = styled.View`
  padding: 25px 30px;
  background-color: white;
  flex-direction: row;
`;

const Content = styled.Text`
  font-size: 15px;
  font-weight: 500;
  color: #687684;
  flex: 1;
  margin-right: 20px;
`;

const Date = styled.Text`
  font-size: 15px;
  font-weight: 800;
  color: #687684;
`;

const Pray = ({data}: {data: PrayType}) => {
  return (
    <Container>
      <Content>{data.content}</Content>
      <Date>{data.weekend}</Date>
    </Container>
  );
};

export default Pray;
