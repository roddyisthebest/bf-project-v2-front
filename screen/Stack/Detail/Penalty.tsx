import React, {useState} from 'react';
import {View, FlatList, SafeAreaView} from 'react-native';
import {PenaltyType} from '../../../types/Penalty';
import PenaltyComponent from '../../../components/Penalty';
const Penalty = () => {
  const [data, setData] = useState<PenaltyType[]>([
    {
      id: 1,
      User: {
        name: '배성연',
        id: 1,
        img: 'https://k.kakaocdn.net/dn/dpk9l1/btqmGhA2lKL/Oz0wDuJn1YV2DIn92f6DVK/img_110x110.jpg',
        oauth: 'KAKAO',
        Pray: [],
        payed: false,
      },
      paper: 0,
      weekend: '2022-06-05',
    },
    {
      id: 2,
      User: {
        name: '배성연',
        id: 1,
        img: 'https://k.kakaocdn.net/dn/dpk9l1/btqmGhA2lKL/Oz0wDuJn1YV2DIn92f6DVK/img_110x110.jpg',
        oauth: 'KAKAO',
        Pray: [],
        payed: false,
      },
      paper: 1000,
      weekend: '2022-05-21',
    },
  ]);

  const renderItem = ({item}: {item: PenaltyType}) => (
    <PenaltyComponent data={item} />
  );
  return (
    <SafeAreaView style={{flex: 1}}>
      <FlatList
        data={data}
        ItemSeparatorComponent={() => (
          <View style={{height: 1, backgroundColor: '#ced5dc'}} />
        )}
        renderItem={renderItem}
        keyExtractor={item => item.id.toString()}
      />
    </SafeAreaView>
  );
};

export default Penalty;
