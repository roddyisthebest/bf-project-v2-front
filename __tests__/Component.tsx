import 'react-native';
import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import Penalty from '../components/Penalty';
import {PenaltyType} from '../types/Penalty';
import PenaltyEditable from '../components/PenaltyEditable';
import {User} from '../types/User';
import {PrayType} from '../types/Pray';
import Pray from '../components/Pray';
import PrayEditable from '../components/PrayEditable';
import {TweetType} from '../types/Tweet';
import Tweet from '../components/Tweet';
import {useSelector} from 'react-redux';

jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');
jest.mock('react-redux');
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

describe('Penalty', () => {
  it('스냅 샷', () => {
    const data: PenaltyType = {
      id: 1,
      paper: 1000,
      weekend: '2022-01-01',
      User: {
        id: 1,
        oauth: 'KAKAO',
        name: 'hyojin',
        img: 'https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef',
        payed: true,
        Prays: null,
        Penalties: [],
        Followers: [],
        Followings: [],
        Service: {},
        createdAt: '2022-09-05 11:30:02',
        updatedAt: '2022-09-05 11:30:02',
      },
    };
    render(<Penalty data={data} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});

describe('PenaltyEditable', () => {
  it('스냅 샷', () => {
    const data: User = {
      id: 1,
      oauth: 'KAKAO',
      name: 'hyojin',
      img: 'https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef',
      payed: true,
      Prays: null,
      Penalties: [{paper: 1000, id: 1, weekend: '2022-06-01', User: {}}],
      Followers: [],
      Followings: [],
      Service: {
        tweet: false,
        penalty: false,
        pray: false,
        UserId: 231,
        id: 1,
      },
      createdAt: '2022-09-05 11:30:02',
      updatedAt: '2022-09-05 11:30:02',
    };
    render(<PenaltyEditable data={data} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});

describe('Pray', () => {
  it('스냅샷', () => {
    const data: PrayType = {
      id: 1,
      content: 'hello hyojin',
      weekend: '2022-09-05 11:30:02',
    };
    render(<Pray data={data} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});

describe('PrayEditable', () => {
  it('스냅샷', () => {
    const data: User = {
      id: 1,
      oauth: 'KAKAO',
      name: 'hyojin',
      img: 'https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef',
      payed: true,
      Prays: null,
      Penalties: [],
      Followers: [],
      Followings: [],
      Service: {tweet: false, penalty: false, pray: false, UserId: 231, id: 1},
      createdAt: '2022-09-05 11:30:02',
      updatedAt: '2022-09-05 11:30:02',
    };
    render(<PrayEditable data={data} editable={true} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});

describe('Tweet', () => {
  it('스냅샷', () => {
    const useSelectorCopy = useSelector as jest.Mock<any>;
    useSelectorCopy.mockImplementation(selector =>
      selector({
        userInfo: {
          img: 'https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef',
          id: 23,
        },
      }),
    );
    const data: TweetType = {
      id: 1,
      img: 'https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef',
      content: 'hyojin saranghae',
      User: {
        id: 1,
        oauth: 'KAKAO',
        name: 'hyojin',
        img: 'https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef',
        payed: true,
        Prays: null,
        Penalties: [],
        Followers: [],
        Followings: [],
        Service: {
          tweet: false,
          penalty: false,
          pray: false,
          UserId: 231,
          id: 1,
        },
        createdAt: '2022-09-05 11:30:02',
        updatedAt: '2022-09-05 11:30:02',
      },
      createdAt: '2022-09-05 11:30:02',
      updatedAt: '2022-09-05 11:30:02',
    };
    render(<Tweet data={data} del={() => {}} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
  it('버튼 클릭시 함수 실행', () => {
    const useSelectorCopy = useSelector as jest.Mock<any>;
    useSelectorCopy.mockImplementation(selector =>
      selector({
        userInfo: {
          img: 'https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef',
          id: 23,
        },
      }),
    );
    const data: TweetType = {
      id: 1,
      img: 'https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef',
      content: 'hyojin saranghae',
      User: {
        id: 23,
        oauth: 'KAKAO',
        name: 'hyojin',
        img: 'https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef',
        payed: true,
        Prays: null,
        Penalties: [],
        Followers: [],
        Followings: [],
        Service: {
          tweet: false,
          penalty: false,
          pray: false,
          UserId: 231,
          id: 1,
        },
        createdAt: '2022-09-05 11:30:02',
        updatedAt: '2022-09-05 11:30:02',
      },
      createdAt: '2022-09-05 11:30:02',
      updatedAt: '2022-09-05 11:30:02',
    };
    const handleClick = jest.fn();
    render(<Tweet data={data} del={handleClick} />);
    let button = screen.getByTestId('button');
    expect(button).toHaveTextContent('삭제하기');
    fireEvent(button, 'click');
    expect(handleClick).toBeCalled();
  });
});
