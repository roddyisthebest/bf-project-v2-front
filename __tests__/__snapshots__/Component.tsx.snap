// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Penalty 스냅 샷 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 25,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 25,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {
          "marginBottom": 12.5,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 15,
            "fontWeight": "800",
          },
        ]
      }
    >
      금액
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#687684",
            "fontSize": 15,
            "fontWeight": "800",
          },
        ]
      }
    >
      1000
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 15,
            "fontWeight": "800",
          },
        ]
      }
    >
      날짜
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#687684",
            "fontSize": 15,
            "fontWeight": "800",
          },
        ]
      }
    >
      2022-01-01
    </Text>
  </View>
</View>
`;

exports[`PenaltyEditable 스냅 샷 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 25,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 25,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 20,
        "opacity": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "height": 30,
            "marginRight": 15,
            "width": 30,
          },
        ]
      }
    >
      <Image
        borderRadius={50}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 30,
              "width": 30,
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 22.5,
            "fontWeight": "800",
          },
        ]
      }
    >
      hyojin
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {
          "marginBottom": 12.5,
          "marginTop": 5,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 15,
            "fontWeight": "800",
          },
        ]
      }
    >
      금액
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#687684",
            "fontSize": 15,
            "fontWeight": "800",
          },
        ]
      }
    >
      1000
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 15,
            "fontWeight": "800",
          },
        ]
      }
    >
      payed
    </Text>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#DFF3F1",
          "borderRadius": 20,
          "height": 30,
          "justifyContent": "center",
          "opacity": 1,
          "width": 30,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#10DDC2",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`Pray 스냅샷 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flexDirection": "row",
        "paddingBottom": 25,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 25,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#687684",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontSize": 15,
          "fontWeight": "500",
          "marginRight": 20,
        },
      ]
    }
  >
    hello hyojin
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#687684",
          "fontSize": 15,
          "fontWeight": "800",
        },
      ]
    }
  >
    2022-09-05 11:30:02
  </Text>
</View>
`;

exports[`PrayEditable 스냅샷 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 15,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 20,
        "opacity": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "height": 30,
            "marginRight": 15,
            "width": 30,
          },
        ]
      }
    >
      <Image
        borderRadius={50}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 30,
              "width": 30,
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 22.5,
            "fontWeight": "800",
          },
        ]
      }
    >
      hyojin
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 20,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 20,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#10ddc2",
          "borderRadius": 50,
          "height": 30,
          "justifyContent": "center",
          "marginBottom": 0,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 0,
          "opacity": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 12,
              "fontWeight": "700",
            },
          ]
        }
      >
        추가하기
      </Text>
    </View>
  </View>
</View>
`;

exports[`Tweet 스냅샷 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 40,
      },
      Object {
        "minHeight": 500,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginRight": 20,
          "width": 50,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "borderColor": "#eff5fb",
            "borderRadius": 50,
            "borderStyle": "solid",
            "borderWidth": 1,
            "width": 52,
          },
        ]
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Array [
            Object {
              "height": 50,
              "width": 50,
            },
          ]
        }
      >
        <Image
          borderRadius={50}
          resizeMode="cover"
          source={
            Object {
              "uri": "https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 50,
                "width": 50,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "800",
            },
          ]
        }
      >
        hyojin
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#687684",
              "fontSize": 12,
              "fontWeight": "500",
              "marginBottom": 0,
              "marginLeft": 7.5,
              "marginRight": 7.5,
              "marginTop": 0,
            },
          ]
        }
      >
        @
        KAKAO
         · 
        11 days ago
      </Text>
    </View>
    <TextInput
      editable={false}
      multiline={true}
      numberOfLines={10}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#ced5dc",
            "borderStyle": "solid",
            "color": "black",
            "height": 80,
            "marginTop": 10,
          },
          Object {
            "textAlignVertical": "top",
          },
        ]
      }
      value="hyojin saranghae"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#e5e5e5",
          "borderRadius": 30,
          "borderWidth": 0.5,
          "marginTop": 30,
          "opacity": 1,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 280,
            },
          ]
        }
      >
        <Image
          borderRadius={30}
          defaultSource={
            Object {
              "testUri": "../../../assets/img/loading.png",
            }
          }
          resizeMode="cover"
          source={
            Object {
              "uri": "undefined/https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-15/296105988_735019344242739_8809340187758356692_n.jpg?stp=dst-jpg_e35_p480x480&_nc_ht=scontent-ssn1-1.cdninstagram.com&_nc_cat=103&_nc_ohc=8FKKENxkJq0AX_C-dmH&tn=BEmipNgrm38iVWSi&edm=ALQROFkBAAAA&ccb=7-5&ig_cache_key=Mjg5Mjg3OTkzOTcxNjI2OTk2Nw%3D%3D.2-ccb7-5&oh=00_AT--zRwVOwpBF0Q2WOcAZMTTTZdHO2xVJvDleUXOK-uKGw&oe=63214EBE&_nc_sid=30a2ef",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 280,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
